### PR TITLE
Split out options & model in props

### PR DESCRIPTION
**Pull request by ccolgrove**
Core `monaco-editor` tries to deep copy the new `options` on update, but when the `model` is an option it fails because the model contains a recursive data structure (doubly-linked list) that causes an infinite loop. (upstream issue: https://github.com/microsoft/monaco-editor/issues/2027)

Fortunately, once you've set the model you never need to update it. This PR separates out the model from the options so `componentDidUpdate` only passes in options that are actually copyable.

There are a lot of ways to implement this so definitely happy for any feedback on the code.

### CODE REVIEW

**Comment by domoritz** [comment]
Thank you for the pull request. Please make sure that the tests pass.

**Comment by ccolgrove** [comment]
thanks @domoritz - tests should pass now

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "react-monaco-editor",
+  "name": "@hex-inc/react-monaco-editor",
```
Comment (domoritz) [inline package.json:2]:
```suggestion
  "name": "react-monaco-editor",
```

File: package.json [hunk 2]
```diff
@@ -58,7 +59,7 @@
   },
   "dependencies": {
     "@types/react": "^16.x",
-    "monaco-editor": "*",
+    "monaco-editor": "0.19.3",
```
Comment (domoritz) [inline package.json:62]: This makes the module incompatible with 0.20.

File: test/typescript/editor.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ class App extends React.Component<{}, AppState> {
         language="javascript"
         theme="vs-dark"
         value={code}
-        options={{ selectOnLineNumbers: true }}
+        optionsWithoutModel={{ selectOnLineNumbers: true }}
```
Comment (domoritz) [inline test/typescript/editor.tsx:47]: Wouldn't this be a breaking change?

Comment (ccolgrove) [reply test/typescript/editor.tsx:47]: yes. I can make it non-breaking if you'd like - basically we can leave the argument as `options` but just not copy the model if it's passed in additional times.

File: package.json (hunk 2 above)
Comment (ccolgrove) [reply package.json:62]: whoops, this was left in there from experimenting I was doing, will change back

File: package.json (hunk 1 above)
Comment (ccolgrove) [reply package.json:2]: same as below, sorry about that!

**Comment by ccolgrove** [comment]
good for another look @domoritz - thanks!

**Pull request merged**